core/dbt/task/serve.py

```diff
@@ -1,46 +1,28 @@
-import shutil
 import os
+import shutil
+import socketserver
 import webbrowser
-
-from dbt.include.global_project import DOCS_INDEX_FILE_PATH
 from http.server import SimpleHTTPRequestHandler
-from socketserver import TCPServer
-from dbt.events.functions import fire_event
-from dbt.events.types import ServingDocsPort, ServingDocsAccessInfo, ServingDocsExitInfo, EmptyLine
 
+import click
+
+from dbt.include.global_project import DOCS_INDEX_FILE_PATH
 from dbt.task.base import ConfiguredTask
 
 
 class ServeTask(ConfiguredTask):
     def run(self):
         os.chdir(self.config.target_path)
-
-        port = self.args.port
-        address = "0.0.0.0"
-
         shutil.copyfile(DOCS_INDEX_FILE_PATH, "index.html")
 
-        fire_event(ServingDocsPort(address=address, port=port))
-        fire_event(ServingDocsAccessInfo(port=port))
-        fire_event(EmptyLine())
-        fire_event(EmptyLine())
-        fire_event(ServingDocsExitInfo())
-
-        # mypy doesn't think SimpleHTTPRequestHandler is ok here, but it is
-        httpd = TCPServer(  # type: ignore
-            (address, port), SimpleHTTPRequestHandler  # type: ignore
-        )  # type: ignore
-
-        if self.args.open_browser:
-            try:
-                webbrowser.open_new_tab(f"http://127.0.0.1:{port}")
-            except webbrowser.Error:
-                pass
+        port = self.args.port
 
-        try:
-            httpd.serve_forever()  # blocks
-        finally:
-            httpd.shutdown()
-            httpd.server_close()
+        if self.args.browser:
+            webbrowser.open_new_tab(f"http://localhost:{port}")
 
-        return None
+        with socketserver.TCPServer(("", port), SimpleHTTPRequestHandler) as httpd:
+            click.echo(f"Serving docs at {port}")
+            click.echo(f"To access from your browser, navigate to: http://localhost:{port}")
+            click.echo("\n\n")
+            click.echo("Press Ctrl+C to exit.")
```

Comment:
This change makes sense to me. `docs serve` is CLI-only functionality; it might make more sense to just `click.echo`, rather than firing a real event.

If we did want these to keep being "real" events / log messages, we could use the `Formatting` ("FYI") message type that @peterallenwebb added in https://github.com/dbt-labs/dbt-core/pull/6691. I leave it up to your discretion.

In any case, I think we could also just remove the one-off event types that are no longer being used anywhere else in the codebase: `ServingDocsPort`, `ServingDocsAccessInfo`, `ServingDocsExitInfo `